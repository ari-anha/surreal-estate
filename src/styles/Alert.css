.Alert {
  border-radius: 15px;
}

.Alert.alert-success {
  background-color: green;
}

.Alert.alert-error {
  background-color: red;
}
