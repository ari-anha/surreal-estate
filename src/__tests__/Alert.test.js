import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

test("displayes an error message", () => {
  //   const component = render(<Alert message="Error!" />);
  //   const alertMessageNode = component.getByText("Error!");
  //   expect(alertMessageNode.textContent).toBe("Error!");

  // Below is a cleaner, destructured way of writing

  const { getByText, asFragment } = render(<Alert message="Error!" />);

  expect(getByText(/Error/).textContent).toBe("Error!");
  expect(asFragment()).toMatchSnapshot();
});

test("displayes a success message", () => {
  const { getByText, asFragment } = render(
    <Alert message="Success!!!" success />
  );
  //  short for success={true}
  expect(getByText(/Success/).textContent).toBe("Success!!!");
  expect(asFragment()).toMatchSnapshot();
});

test("does not render error or success message if message prop is empty", () => {
  const { asFragment } = render(<Alert message="" />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});
