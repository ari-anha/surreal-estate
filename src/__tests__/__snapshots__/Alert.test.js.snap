// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displayes a success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-success"
  >
    Success!!!
  </div>
</DocumentFragment>
`;

exports[`displayes an error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`does not render error or success message if message prop is empty 1`] = `<DocumentFragment />`;
